src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/wrapper/ErrorHandlingWrapper.java

```diff
@@ -40,15 +41,16 @@ public ErrorHandlingWrapper(ErrorHandler errorHandler) {
         }
     }
 
-    public void wrapDocUploadFailure(Envelope envelope, Supplier<Void> supplier) {
+    public Boolean wrapDocUploadFailure(Envelope envelope, Supplier<Boolean> supplier) {
         try {
-            supplier.get();
+            return supplier.get();
```

Comment:
perhaps returning `true` could be here?
( operation succeeded -> always return true)

Reply:
instead of always returning true in the supplier passed to the wrapper

Reply:
The supplier should return something anyway because of its contract for what I can see, so I am not sure what you suggest would make it any more readable